incubator/hnc/internal/reconcilers/hierarchy_config_test.go

```diff
@@ -31,6 +31,31 @@ var _ = Describe("Hierarchy", func() {
 		Eventually(hasChild(ctx, barName, fooName)).Should(Equal(true))
 	})
 
+	It("should remove the hierarchyconfiguration singleton in an excluded namespacee", func() {
+		// Set the excluded-namespace "kube-system"'s parent to "bar".
+		config.ExcludedNamespaces = map[string]bool{"kube-system": true}
+		exHier := newHierarchy("kube-system")
+		exHier.Spec.Parent = barName
+		updateHierarchy(ctx, exHier)
+
+		// Verify the hierarchyconfiguration singleton is deleted. Give apiserver
+		// 0.5 second here to be able to process the deletion of the singleton
+		// before we constantly retry getting the singleton with Get().
+		time.Sleep(500 * time.Millisecond)
+		Eventually(canGetHierarchy(ctx, "kube-system")).ShouldNot(Succeed())
```

Comment:
`canGetHierarchy` should return a function that `Eventually` can retry, not an answer that never changes (e.g. see `hasCondition`). Then you get get rid of `Sleep`.

As a general rule, use `Expect()` and not `Eventually()` if you're not truly retrying.